gateway/slack.go

```diff
@@ -226,6 +237,41 @@ func (sc *SlackClient) ResolveNickToUser(nick string) *SlackUser {
 	return nil
 }
 
+// ResolveDMToUser resolves a DM/IM Channel ID to the User the DM is for
+func (sc *SlackClient) ResolveDMToUser(dmChannelID string) (*SlackUser, error) {
+	sc.RLock()
```

Comment:
not actually sure if we need to grab the read lock here. if we do, then `ResolveChannel` and `ResolveUser` above should also grab the read lock.